### PR TITLE
Updated to include foam_week_year

### PR DESCRIPTION
**Pull request by ChThH**
**Add FOAM_DATE_WEEK_YEAR variable**
Adds support for FOAM_DATE_WEEK_YEAR to complement the existing FOAM_DATE_WEEK variable.
This variable returns the ISO 8601 week-numbering year, which can differ from the calendar year for dates at the start/end of the year. For example:

December 30, 2024 (Monday) is in week 1 of 2025, so FOAM_DATE_WEEK_YEAR returns 2025
January 1, 2023 (Sunday) is in week 52 of 2022, so FOAM_DATE_WEEK_YEAR returns 2022

This enables proper ISO week date formatting: ${FOAM_DATE_WEEK_YEAR}-W${FOAM_DATE_WEEK}
Changes:

Added FOAM_DATE_WEEK_YEAR to knownFoamVariables set
Implemented resolver logic following ISO 8601 week date rules
Added tests for edge cases where week year differs from calendar year

### CODE REVIEW

File: packages/foam-vscode/src/services/variable-resolver.spec.ts [hunk 1]
```diff
@@ -265,6 +265,77 @@ describe('variable-resolver, variable resolution', () => {
     });
   });
 
+  it('should resolve FOAM_DATE_WEEK_YEAR correctly for edge cases', async () => {
+    // Test cases where week year differs from calendar year
+    const testCases = [
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/services/variable-resolver.spec.ts]: you can use `test.each`, search it in the codebase for usage examples

**Comment by ChThH** [comment]
Thanks @riccardoferretti I've updated the tests and documentation. Let me know if I did it correctly

File: packages/foam-vscode/src/dated-notes.spec.ts [hunk 2]
```diff
@@ -111,6 +111,7 @@ Month: \${FOAM_DATE_MONTH} (name: \${FOAM_DATE_MONTH_NAME}, short: \${FOAM_DATE_
 Date: \${FOAM_DATE_DATE}
 Day: \${FOAM_DATE_DAY_NAME} (short: \${FOAM_DATE_DAY_NAME_SHORT})
 Week: \${FOAM_DATE_WEEK}
+Week Year: \${FOAM_DATE_WEEK}
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/dated-notes.spec.ts]: using the wrong variable

Comment (riccardoferretti) [reply packages/foam-vscode/src/dated-notes.spec.ts]: (also, you should probably add this to the expectation, otherwise we might not add it at all :) )

File: docs/user/features/templates.md [hunk 3]
```diff
@@ -258,6 +258,7 @@ Supported variables include:
 - `FOAM_DATE_YEAR`: 4-digit year (e.g. 2025)
 - `FOAM_DATE_MONTH`: 2-digit month (e.g. 09)
 - `FOAM_DATE_WEEK`: ISO 8601 week number (e.g. 37)
+- `FOAM_DATE_WEEK_YEAR`: ISO 8601 week-numbering year (e.g. 2024) (may vary from calendar year near Jan 1)
```
Comment (riccardoferretti) [inline docs/user/features/templates.md]:
```suggestion
- `FOAM_DATE_WEEK_YEAR`: the year of the ISO 8601 week number. The year that contains the Thursday of the current week, may vary from calendar year near Jan 1. Often used with `FOAM_DATE_WEEK`.
```

**Review by riccardoferretti — APPROVED** [review]

File: packages/foam-vscode/src/dated-notes.spec.ts (hunk 2 above)
Comment (ChThH) [reply packages/foam-vscode/src/dated-notes.spec.ts]: Thanks for walking me through the process. I've either not done a pull request before or it's been so long I've forgotten (besides that little one earlier).

Comment (riccardoferretti) [reply packages/foam-vscode/src/dated-notes.spec.ts]: Always happy to support, and thanks again for your contributions to the project!
